linked_list/linked_list.py

```diff
@@ -86,10 +145,15 @@ def visit(self):
 
     # method to reverse the singly linked list
     # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def reverse(self):
-        pass
+        cur = self.head
+        last_made = None
+        while cur != None:
+            last_made = Node(cur.value, last_made)
+            cur = cur.next
+        self.head = last_made

```

Comment:
hmm we shouldn't be adding nodes, only reversing the pointers so the last node is now the head and vice versa.

It's commonly done with three pointers: previous, current, and future node